ultralytics/solutions/analytics.py

```diff
@@ -0,0 +1,197 @@
+                (23, 190, 207),
+            ]
+            self.color_palette = [(r / 255, g / 255, b / 255, 1) for r, g, b in color_palette]
+            self.color_cycle = cycle(self.color_palette)
+            self.color_mapping = {}
+
+            # Ensure pie chart is circular
+            self.ax.axis("equal") if type == "pie" else None
+
+        # Set common axis properties
+        self.ax.set_title(self.title, color=self.fg_color, fontsize=fontsize)
+        self.ax.set_xlabel(x_label, color=self.fg_color, fontsize=fontsize - 3)
+        self.ax.set_ylabel(y_label, color=self.fg_color, fontsize=fontsize - 3)
+        self.ax.tick_params(axis="both", colors=self.fg_color)
+
+    def update_line(self, frame_number, total_counts):
+        """
+        Update the line graph with new data.
+
+        Args:
+            frame_number (int): The current frame number.
+            total_counts (int): The total counts to plot.
+        """
+
+        # Update line graph data
+        x_data = self.line.get_xdata()
+        y_data = self.line.get_ydata()
+        x_data = np.append(x_data, float(frame_number))
+        y_data = np.append(y_data, float(total_counts))
+        self.line.set_data(x_data, y_data)
+        self.ax.relim()
+        self.ax.autoscale_view()
+        self.canvas.draw()
+        im0 = np.array(self.canvas.renderer.buffer_rgba())
+        im0 = cv2.cvtColor(im0[:, :, :3], cv2.COLOR_RGBA2BGR)
+
+        # Display and save the updated graph
+        cv2.imshow(self.title, im0) if self.view_img else None
+        self.writer.write(im0) if self.save_img else None
+
+    def update_bar(self, count_dict):
+        """
+        Update the bar graph with new data.
+
+        Args:
+            count_dict (dict): Dictionary containing the count data to plot.
+        """
+
+        # Update bar graph data
+        self.ax.clear()
+        self.ax.set_facecolor(self.bg_color)
+        labels = list(count_dict.keys())
+        counts = list(count_dict.values())
+
+        # Map labels to colors
+        for label in labels:
+            if label not in self.color_mapping:
+                self.color_mapping[label] = next(self.color_cycle)
+
+        colors = [self.color_mapping[label] for label in labels]
+
+        bars = self.ax.bar(labels, counts, color=colors)
+        for bar, count in zip(bars, counts):
+            self.ax.text(
+                bar.get_x() + bar.get_width() / 2,
+                bar.get_height(),
+                str(count),
+                ha="center",
+                va="bottom",
+                color=self.fg_color,
+            )
+
+        # Display and save the updated graph
+        canvas = FigureCanvas(self.fig)
+        canvas.draw()
+        buf = canvas.buffer_rgba()
+        im0 = np.asarray(buf)
+        im0 = cv2.cvtColor(im0, cv2.COLOR_RGBA2BGR)
+
+        self.writer.write(im0) if self.save_img else None
+        cv2.imshow(self.title, im0) if self.view_img else None
```

Comment:
@RizwanMunawar not sure if this is the best way to do this. See this code for an example I did earlier for FastSAM writing/saving:

https://github.com/ultralytics/ultralytics/blob/1a4ac2c6baafd483377d54d0eeb56c91ac440d6d/ultralytics/models/fastsam/prompt.py#L172-L177